jwst/outlier_detection/outlier_detection_tso.py

```diff
@@ -0,0 +1,173 @@
+import numpy as np
+from .outlier_detection import OutlierDetection, _remove_file
+from jwst.resample.resample_utils import build_driz_weight as build_weight
+
+from jwst import datamodels as dm
+import logging
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+__all__ = ["OutlierDetectionTSO"]
+
+
+class OutlierDetectionTSO(OutlierDetection):
+    """Class to flag outlier pixels in DQ of TSO data. Works similarly to
+    imaging outlier detection, but does not resample and uses a rolling median."""
+
+    def __init__(self, input_models: dm.ModelContainer, **pars):
+        """Initialize class for TSO data processing.
+
+        Parameters
+        ----------
+        input_models : ~jwst.datamodels.ModelContainer, str
+            list of data models as ModelContainer or ASN file,
+            one data model for each input 2-D ImageModel
+        """
+        super().__init__(input_models, **pars)
+        self._convert_inputs()
+
+
+    def do_detection(self):
+
+        """Flag outlier pixels in DQ of input images."""
+        weighted_models = self.weight_no_resample()
+
+        maskpt = self.outlierpars.get('maskpt', 0.7)
+        weight_thresholds = self.compute_weight_threshold(weighted_models, maskpt)
+
+        rolling_width = self.outlierpars.get('n_ints', 25)
+        if (rolling_width > 1) and (rolling_width < len(weighted_models)):
+            medians = self.compute_rolling_median(weighted_models, weight_thresholds, w = rolling_width)
+
+        else:
+            medians = super().create_median(weighted_models)
+            # this is a 2-D array, need to repeat it into the time axis
+            # for consistent shape with rolling median case
+            shp = (len(weighted_models), medians.shape[0], medians.shape[1])
+            medians = np.broadcast_to(medians, shp)
+
+        # Save median model if pars['save_intermediate_results'] is True
+        # this will be a CubeModel with rolling median values.
+        if self.outlierpars['save_intermediate_results']:
+            median_model = dm.CubeModel(data=medians)
+            with dm.open(weighted_models[0]) as dm0:
+                median_model.update(dm0)
+            self.save_median(median_model) 
+        else:
+            # since we're not saving intermediate results,
+            # remove those files
+            if not self.outlierpars['in_memory']:
+                for fn in weighted_models._models:
+                    _remove_file(fn)
+
+        # no need for blotting, resample is turned off for TSO
+        blot_models = dm.ModelContainer(open_models=False)
```

Comment:
```suggestion
        blot_models = dm.ModelContainer()
```
Although it won't cause an error `open_models` isn't understood by `ModelContainer`.